# $NetBSD$
#
###########################################################
#                  Generated by fbsd2pkg                  #
#              Fri Sep 14 20:48:17 CDT 2018               #
###########################################################

###########################################################
# Unconverted and partially converted FreeBSD port syntax:

# Trinity core programs: Inchworm, Chrysalis, Butterfly
# Bundled plugins: Trinity is very sensitive to having the correct versions
# of some of these.  Test thoroughly if you change any of them.
# To-do:
#	Unbundle collectl (low priority)
## Should work on other 64-bit archs, but untested
#ONLY_FOR_ARCH=	amd64

# Should salmon use bl3?
# bowtie should not have bl3
DEPENDS=	coreutils>=0:../../sysutils/coreutils
DEPENDS+=	bash>=0:../../shells/bash
DEPENDS+=	slclust>=0:../../wip/slclust
DEPENDS+=	salmon>=0:../../wip/salmon
DEPENDS+=	bowtie>=0:../../wip/bowtie
DEPENDS+=	bowtie2>=0:../../wip/bowtie2
DEPENDS+=	samtools>=1.3:../../biology/samtools
DEPENDS+=	jellyfish>=2.1.4:../../wip/jellyfish
DEPENDS+=	parafly>=0:../../wip/parafly
DEPENDS+=	fastool>=0:../../wip/fastool
DEPENDS+=	p5-transdecoder>=0:../../wip/p5-transdecoder
DEPENDS+=	Trimmomatic>=0.38:../../wip/trimmomatic
DEPENDS+=	rsem>=1.2.27:../../wip/rsem

DISTNAME=	trinity-${PV}
CATEGORIES=	biology
MASTER_SITES=	${MASTER_SITE_GITHUB:=trinityrnaseq/}
GITHUB_PROJECT=	trinityrnaseq
GITHUB_TAG=	Trinity-v${PV}

MAINTAINER=	bacon@NetBSD.org
HOMEPAGE=	http://trinityrnaseq.github.io/
COMMENT=	Assembles transcript sequences from Illumina RNA-Seq data
LICENSE=	modified-bsd

# MAKE_JOBS_SAFE=	no

USE_LANGUAGES=	c c++
USE_JAVA=	run
USE_TOOLS+=	cmake gmake pax
USE_CMAKE=	yes

REPLACE_BASH=	util/*/*.sh
REPLACE_BASH+=	util/misc/alt_GG_read_partitioning_JCornish/genwig.sh
REPLACE_BASH+=	Analysis/DifferentialExpression/cluster_sample_data/runMe.sh
REPLACE_PERL=	util/*/*.pl Chrysalis/analysis/*.pl 
REPLACE_PYTHON=	util/support_scripts/trinity_installer.py
REPLACE_PYTHON+=util/misc/alt_GG_read_partitioning_JCornish/genwig2.py
REPLACE_PYTHON+=util/misc/sim_test_framework/run_Trinity_eval.sh
REPLACE_PYTHON+=util/misc/TPM_weighted_gene_length.py
REPLACE_PYTHON+=Analysis/SuperTranscripts/*.py
REPLACE_PYTHON+=Analysis/SuperTranscripts/*/*.py

SUBST_CLASSES+=		compiler
SUBST_STAGE.compiler=	post-patch
SUBST_SED.compiler+=	-e "s|gcc|${CC}|g"
SUBST_SED.compiler+=	-e "s|g++|${CXX}|g"
SUBST_FILES.compiler+=	${WRKSRC}/Chrysalis/Makefile*

SUBST_CLASSES+=		paths
SUBST_STAGE.paths=	post-patch
SUBST_SED.paths+=	-e 's|$$FindBin::Bin|$$FindBin::Bin/../libexec/trinity|g'
SUBST_SED.paths+=	-e 's|$$FindBin::RealBin|$$FindBin::RealBin/../libexec/trinity|g'
SUBST_SED.paths+=	-e 's|$$ROOTDIR/trinity-plugins/BIN/ParaFly|ParaFly|g'
SUBST_SED.paths+=	-e 's|$$ROOTDIR/trinity-plugins/Trimmomatic/trimmomatic.jar|${JAVAJARDIR}/trimmomatic.jar|g'
SUBST_SED.paths+=	-e 's|$$ROOTDIR/trinity-plugins/Trimmomatic|${LOCALBASE}/share/Trimmomatic|g'
SUBST_FILES.paths+=	${WRKSRC}/Trinity

SUBST_CLASSES+=		jelly
SUBST_STAGE.jelly=	post-patch
SUBST_SED.jelly+=	-e 's|$$JELLYFISH_DIR/bin/jellyfish|jellyfish|g'
SUBST_FILES.jelly+=	${WRKSRC}/util/misc/run_jellyfish.pl

SUBST_CLASSES+=		trimmo
SUBST_STAGE.trimmo=	post-patch
SUBST_SED.trimmo+=	-e 's|$$FindBin::RealBin/../../trinity-plugins/Trimmomatic/trimmomatic.jar|${JAVAJARDIR}/trimmomatic.jar|g'
SUBST_SED.trimmo+=	-e 's|/seq/regev_genome_portal/SOFTWARE/BIN/trimmomatic.jar|${JAVAJARDIR}/trimmomatic.jar|g'
SUBST_FILES.trimmo+=	${WRKSRC}/util/misc/run_trimmomatic_qual_trimming.pl

SUBST_CLASSES+=		rpath
SUBST_STAGE.rpath=	post-patch
SUBST_SED.rpath+=	-e "s|-lm -pthread|-lm -pthread -Wl,-rpath=${_GCC_RUNTIME}|g"
SUBST_FILES.rpath+=	${WRKSRC}/Chrysalis/Makefile

SUBST_CLASSES+=		trinity
SUBST_STAGE.trinity=	post-patch
SUBST_SED.trinity+=	-e 's|$$FindBin::Bin/../../Trinity|Trinity|g'
SUBST_FILES.trinity+=	${WRKSRC}/util/support_scripts/write_partitioned_trinity_cmds.pl

SUBST_CLASSES+=		build
SUBST_STAGE.build=	post-build
SUBST_SED.build+=	-e "s|/usr/bin/perl|${PREFIX}/bin/perl|g"
SUBST_FILES.build+=	${WRKSRC}/trinity-plugins/COLLECTL/collectl/collectl
SUBST_FILES.build+=	${WRKSRC}/trinity-plugins/COLLECTL/collectl/*.pl

CFLAGS+=	-fopenmp
CXXFLAGS=	-fopenmp 
FFLAGS=		-fopenmp 

PV=		2.8.3
EXAMPLESDIR=	${PREFIX}/share/examples/trinity
LIBEXEC_DIR=	${PREFIX}/libexec/trinity
PLUGINS_DIR=	${LIBEXEC_DIR}/trinity-plugins

# Sets OPSYS, OS_VERSION, MACHINE_ARCH, etc..
# .include "../../mk/bsd.prefs.mk"

INSTALLATION_DIRS=	bin include lib ${PKGMANDIR}/man1 share/doc share/examples

# FIXME: More may need to be installed
do-install:
	${INSTALL_SCRIPT} ${WRKSRC}/Trinity ${DESTDIR}${PREFIX}/bin
	@${MKDIR} ${DESTDIR}${LIBEXEC_DIR}/Chrysalis/bin
	${INSTALL_PROGRAM} ${WRKSRC}/Chrysalis/bin/* \
		${DESTDIR}${LIBEXEC_DIR}/Chrysalis/bin
	${INSTALL_SCRIPT} ${WRKSRC}/Chrysalis/analysis/ReadsToComponents.pl \
		${DESTDIR}${LIBEXEC_DIR}/Chrysalis/bin
	@${MKDIR} ${DESTDIR}${LIBEXEC_DIR}/Inchworm/bin
	${INSTALL_PROGRAM} ${WRKSRC}/Inchworm/bin/* \
		${DESTDIR}${LIBEXEC_DIR}/Inchworm/bin
	@${MKDIR} ${DESTDIR}${LIBEXEC_DIR}/Butterfly
	${INSTALL_SCRIPT} ${WRKSRC}/Butterfly/Butterfly.jar \
		${DESTDIR}${LIBEXEC_DIR}/Butterfly
	cd ${WRKSRC} && pax -rw PerlLib \
		${DESTDIR}${LIBEXEC_DIR}
	cd ${WRKSRC} && pax -rw util \
		${DESTDIR}${LIBEXEC_DIR}
	cd ${WRKSRC} && pax -rw Analysis \
		${DESTDIR}${LIBEXEC_DIR}
	@${MKDIR} ${DESTDIR}${PLUGINS_DIR}/scaffold_iworm_contigs
	${INSTALL_PROGRAM} ${WRKSRC}/trinity-plugins/scaffold_iworm_contigs/scaffold_iworm_contigs \
		${DESTDIR}${PLUGINS_DIR}/scaffold_iworm_contigs
	@${MKDIR} ${DESTDIR}${PLUGINS_DIR}/BIN
	${INSTALL_PROGRAM} ${WRKSRC}/trinity-plugins/BIN/seqtk-trinity \
		${DESTDIR}${PLUGINS_DIR}/BIN
	cd ${WRKSRC}/sample_data && pax -rw \* \
		${DESTDIR}${EXAMPLESDIR}
	${RLN} ${DESTDIR}${PREFIX}/bin/Trinity \
		${DESTDIR}${PREFIX}/libexec/trinity/Trinity

.include "../../lang/python/application.mk"
.include "../../biology/htslib/buildlink3.mk"
# CentOS doesn't have zlib in the base, so uncomment if needed.
# .include "../../devel/zlib/buildlink3.mk"
.include "../../mk/bsd.pkg.mk"
